e3nn/o3/_tensor_product/_tensor_product.py

```diff
@@ -440,6 +474,9 @@ def right(self, y, weight: Optional[torch.Tensor] = None):
         """
         assert y.shape[-1] == self._in2_dim, "Incorrect last dimension for y"
 
+        if not hasattr(self, '_compiled_main_right'):
+            raise ValueError("`right` method is not compiled, set `compile_right` to True when creating the TensorProduct")
```

Comment:
not sure that this is torchscript compatable

better would be to generate a tiny `GraphModule` that just throws a useful error for `graphmod_right` if `not compile_right`. This should do it:

```python
graph = fx.Graph()
graph.placeholder("...")  # give it relevant arguments so it doesn't die on wrong number of args
graph.call_function(torch._assert, args=(False, "`right` method is not compiled, set `compile_right` to True when creating the TensorProduct")
```

Reply:
this also eliminates the ugly conditional in init for calling `_codegen_register`

Reply:
it also handles the right-only case correctly (this check is currently missing from forward)

Reply:
done